modules/gui/src/com/haulmont/cuba/gui/components/LookupField.java

```diff
@@ -33,6 +33,7 @@
     String NAME = "lookupField";
 
     TypeToken<LookupField<String>> TYPE_STRING = new TypeToken<LookupField<String>>(){};
+    TypeToken<LookupField<Integer>> TYPE_INTEGER = new TypeToken<LookupField<Integer>>(){};
```

Comment:
Thank you for your intention, but in bugfix issues we do not provide additional abilities :)
Remove it and just use `LookupField#of(Class<T> valueClass)` like this:
```
uiComponents.create(LookupField.of(Integer.TYPE))
```

In general its pretty good, after fixes we will merge it.

Reply:
ok, fixed